packages/core/theme/index.js

```diff
@@ -1,11 +1,10 @@
 import * as base from "./base";
 import * as components from "./components";
-import icons from "./icons";
 import { buildTheme } from "../utils";
 import merge from "lodash.merge";
 
 export const getTheme = (userDefined = {}) => {
   const mergedBase = merge(base, userDefined);
-  const builtBase = buildTheme(components, mergedBase);
-  return merge({ ...builtBase, icons }, userDefined);
+  const builtBase = buildTheme(mergedBase, components);
+  return merge({ ...builtBase }, userDefined);
```

Comment:
I know (a variation on) this code was already there, but can you explain why `userDefined` gets merged into the base theme twice – both before and after the component themes get their values resolved.

Reply:
Relatedly, does this theming system allow for the `userDefined` theme to include component themes?

Reply:
I believe it has something to do with your second question. However, @alkamin would be able to speak to the logic better. 

My interpretation is as follows: 
I think it first merges base theme overrides (non-components). Then the base component theme gets merged into our new base and will utilize the new base theme with overrides. And finally, any custom component themes get merged in last. Kind of hard to explain.

base theme (no components) + base overrides = `newBase`
`newBase` + base component theme = `newBaseWithComps`
`newBaseWithComps` + component overrides = `finalTheme`

Reply:
☝️ This is correct. If we only did the first merge, the default component themes would overwrite the user-defined themes. If we only did the second merge, the base component themes would reference the default base theme values instead of the user-defined theme values.